### ISSUE TITLE
BarricadeManager.dropBarricade Angle Parameters

### ISSUE THREAD

**Issue by UFI8** [SmartlyDressedGames/Unturned-3.x-Community#849]
TLDR: I want to replicate Barricade using BarricadeMgr.DropBarricade and BarricadeData stored in memory, where do I copy the Angle_X, Angle_Y, Angle_Z Data from?

Hello,
Hate to bother with probably stupid question, but I am trying to replicate a barricade for a Magic Carpet, and the rotation/angle is always way off what it was before. I am storing the BarricadeData in memory and then destroying the barricade, then using dropBarricade w/ the same info. The angle is always off though, even when using BarricadeData.Angle_X , Y, Z from the stored Barricade for the Angle_X, Angle_Y, Angle_Z
I have tried almost every rotation, position inside of Barricade data for 2 days with zero success, 

Code I am using
`Barricade barricadetest = new Barricade(newID: bdata.Barricade.id);`
`barricadetest.state = destroyedBarricade.Barricade.state;`
`BarricadeManager.dropBarricade(barricade: barricadetest, hit: barricadeTransform, point: bdata.point, angle_x: bdata.angle_x, angle_y: bdata.angle_y, angle_z: bdata.angle_z, owner: bdata.owner, group: bdata.group);`
[Before](https://i.imgur.com/JVRtID8.jpg)
[After](https://i.imgur.com/FffToGv.jpg)

**Comment by DiFFoZ** [comment]
Try this
`MeasurementTool.byteToAngle(angle_y)`

**Comment by UFI8** [comment]
@DiFFoZ That was one of the things I did try. I just retest it to be sure.
[Before](https://i.imgur.com/FvgsHfU.png)
[After](https://i.imgur.com/tOBLLom.png)
Does not seem it's that easy, unfortunately 

It seems Nelson get rotation (eulerangles) from some method called getRotation
`Vector3 eulerAngles = getRotation(itemBarricadeAsset, angle_x, angle_y, angle_z).eulerAngles;`
	`angle_x = Mathf.RoundToInt(eulerAngles.x / 2f) * 2;`
	`angle_y = Mathf.RoundToInt(eulerAngles.y / 2f) * 2;`
	`angle_z = Mathf.RoundToInt(eulerAngles.z / 2f) * 2;``

I would imagine it take something like Position or actual Rotation from the barricade, but I tried both of those with no luck. Also tryed LocalPosition and LocalRotation.

Only God and Nelson seem to know where the getRotation method gets angles from.

**Comment by Adamv21** [comment]
Decompile and look

**Comment by UFI8** [comment]
The problem is just due to the  Quaternion.Euler's, it seems DropBarricade was made for internal calls or just not really for replicating buildables, it would be nice if we had a bit more control over the Barricade System.
 I guess I'll just have to figure out the exact match and reserve engineer it all, alternatively it would be nice if we had just a plain function for straight up spawning a barricade easily.
I would use BarricadeManager.spawnBarricade and the steam call TellBarricade if spawnbarricade/instancecount weren't private. Oh well

**Comment by SDGNelson** [comment]
Will clean up / make easier for the next update!

**Comment by SDGNelson** [comment]
In the next update this code has been cleaned up somewhat: you can use dropPlantedBarricade or dropNonPlantedBarricade which take quaternions for rotation and handle replication. Similar for structures as well!
